frontend/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -356,23 +373,47 @@ function DataFrame({
 
   // Automatic table height calculation: numRows +1 because of header, and +3 pixels for borders
   let maxHeight = Math.max((numRows + 1) * ROW_HEIGHT + 3, MIN_TABLE_HEIGHT)
-  let height = Math.min(maxHeight, 400)
+  let height = Math.min(maxHeight, DEFAULT_TABLE_HEIGHT)
 
-  if (propHeight) {
+  if (element.height) {
     // User has explicitly configured a height
-    height = Math.max(propHeight, MIN_TABLE_HEIGHT)
-    maxHeight = Math.max(propHeight, maxHeight)
+    height = Math.max(element.height, MIN_TABLE_HEIGHT)
+    maxHeight = Math.max(element.height, maxHeight)
+  }
+
+  if (containerHeight) {
+    // If container height is set (e.g. when used in fullscreen)
+    // The maxHeight and height should not be larger than container height
+    height = Math.min(height, containerHeight)
+    maxHeight = Math.min(maxHeight, containerHeight)
+
+    if (!element.height) {
+      // If no explicit height is set, set height to max height (fullscreen mode)
+      height = maxHeight
+    }
```

Comment:
Just want to clarify - we only want to set `height` equal to `maxHeight` when the user doesn't explicitly set height && there is a `containerHeight`? 
Essentially I'm curious in the case where there is no `containerHeight` and no explicit `height` set (if that exists), would we still want to set `height` equal to `maxHeight`?

Reply:
the sizing is a bit confusing since it covers various scenarios and is also used for things like drag & drop resizing 😅 But yes, the only case the default table height should be set to the max height is when there is a container height (e.g. when in fullscreen mode) and no explicit height. So if you open on fullscreen mode, it fills the full available space. When there is a no container height (in the normal usage), the intention is to limit the height to a maximum value of 400.